### ISSUE TITLE
Libtask error when using PG

### ISSUE THREAD

**Issue by phipsgabler** [TuringLang/Turing.jl#1873]
What's wrong here?

```julia
julia> @model function test(K, N)
           w ~ Dirichlet(K, 1.0)
           μ ~ MvNormal(zeros(K), 10.0I)
           
           x = Vector{Int}(undef, N)
           y = Vector{Float64}(undef, N)
           for k = 1:N
               x[k] ~ Categorical(w)
               y[k] ~ Normal(μ[x[k]], 1.0)
           end
           return (;w, x, y)
       end
test (generic function with 4 methods)

julia> sample(test(2, 10) | (;y = randn(10)), Gibbs(PG(100, :x), HMC(0.05, 10, :w, :μ)), 1000)

Sampling 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| Time: 0:00:02
ERROR: MethodError: no method matching translate!!(::Core.SSAValue, ::Tuple{Symbol, Symbol}, ::Vector{Any}, ::Bool, ::Core.CodeInfo)
Closest candidates are:
  translate!!(::Union{Core.SSAValue, Core.SlotNumber}, ::Core.NewvarNode, ::Vector{Any}, ::Bool, ::Any) at ~/.julia/packages/Libtask/9mVaO/src/tapedfunction.jl:343
  translate!!(::Union{Core.SSAValue, Core.SlotNumber}, ::GlobalRef, ::Vector{Any}, ::Bool, ::Any) at ~/.julia/packages/Libtask/9mVaO/src/tapedfunction.jl:349
  translate!!(::Union{Core.SSAValue, Core.SlotNumber}, ::Core.SlotNumber, ::Vector{Any}, ::Bool, ::Any) at ~/.julia/packages/Libtask/9mVaO/src/tapedfunction.jl:359
  ...
Stacktrace:
  [1] translate!(tape::Vector{Libtask.AbstractInstruction}, ir::Core.CodeInfo)
    @ Libtask ~/.julia/packages/Libtask/9mVaO/src/tapedfunction.jl:320
  ...
```

**Comment by yebai** [comment]
@KDr2 can you take a look at this please?

**Comment by KDr2** [comment]
I got another error on Turing.jl@master and Libtask 0.8.1:

```
Sampling 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| Time: 0:00:01
ERROR: UndefVarError: I not defined                                                                                                                         Stacktrace:                                                                                                                                                   
[1] test(__model__::DynamicPPL.Model{typeof(test), (:K, :N), (), (), Tuple{Int64, Int64}, Tuple{}, 
DynamicPPL.ConditionContext{(:y,), NamedTuple{(:y,), Tuple{Vector{Float64}}}, 
DynamicPPL.DefaultContext}}, 
__varinfo__::DynamicPPL.UntypedVarInfo{DynamicPPL.Metadata{Dict{AbstractPPL.VarName, Int64}, Vector{Distribution}, Vector{AbstractPPL.VarName}, Vector{Real}, Vector{Set{DynamicPPL.Selector}}}, Float64}, __context__::DynamicPPL.SamplingContext{DynamicPPL.SampleFromPrior, DynamicPPL.ConditionContext{(:y,), NamedTuple{(:y,), Tuple{Vector{Float64}}}, DynamicPPL.DefaultContext}, Random._GLOBAL_RNG}, K::Int64, N::Int64)                                                                                                                                                               @ Main ./REPL[2]:3                                                                                                                                        
[2] macro expansion                                                                                                                                           
@ ~/.julia/packages/DynamicPPL/1qg3U/src/model.jl:493 [inlined]                                                                                           
[3] _evaluate!!                                                                                                                                               
@ ~/.julia/packages/DynamicPPL/1qg3U/src/model.jl:476 [inlined]                                                                                           
[4] evaluate_threadunsafe!!                                                                                                                                   
@ ~/.julia/packages/DynamicPPL/1qg3U/src/model.jl:451 [inlined]                                                                                           
[5] evaluate!!                                                                                                                                                
@ ~/.julia/packages/DynamicPPL/1qg3U/src/model.jl:404 [inlined]                        
```

Is there a way to reproduce your error?

**Comment by devmotion** [comment]
Did you load LinearAlgebra?

**Comment by KDr2** [comment]
> Did you load LinearAlgebra?

Thank you, I got it after `using LinearAlgebra`.

It's caused by the new syntax to construct NamedTuple `(; w, x, y)`, I will try to find a solution.

For now, we can use an ordinary tuple to avoid this: `return (w, x, y)`.

**Comment by yebai** [comment]
This is now fixed in Libtask. But the fix will take effect until #1858 is merged, which is currently upper bounding Libtask to a previous version (`v0.7`).
